src/validate-and-json.js

```diff
@@ -0,0 +1,80 @@
+// @flow
+import rootRelativePath from "./root-relative-path.js";
+import generateBrokenEdgeMap from "./generate-broken-edge-map.js";
+
+import type {MarkerCache, Options, JsonItem, MarkerEdge} from "./types.js";
+
+const reportBrokenEdge = (
+    options: Options,
+    sourceFile: string,
+    mappedFix: {
+        fix: string,
+        edge: MarkerEdge,
+    },
+    jsonItems: Array<JsonItem>,
+): void => {
+    const {
+        markerID,
+        sourceLine,
+        targetLine,
+        targetFile,
+        sourceChecksum,
+        targetChecksum,
+    } = mappedFix.edge;
+
+    if (targetLine == null || targetChecksum == null) {
```

Comment:
No change requested here:

Seeing some of these validators and their similarities, a future refactor might be warranted that mitigates some of the duplication a bit. Not sure what that would look like without more thought and without adding yet more looping. Just something to think about for now.